File: sprint-service/src/main/java/ma/ehei/projectsprint/sprintservice/entities/Sprint.java
package ma.ehei.projectsprint.sprintservice.entities;

import jakarta.persistence.*;
import lombok.*;
import ma.ehei.projectsprint.sprintservice.model.Project;

import java.io.Serializable;
import java.time.LocalDate;
import java.util.Date;

@Getter @Setter @ToString @NoArgsConstructor @AllArgsConstructor @Builder
@Entity
public class Sprint implements Serializable {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String sprintName;
    private String sprintGoal;
    private LocalDate dateDebut;
    private Date dateFin;
    @Transient
    private Project project;
    private Integer projectId;
}

File: angular-front-end/src/app/sprints/Models/sprint.ts
import {Project} from "../../projects/Models/project";

export class Sprint {
  id:number=0;
  sprintName:string="";
  sprintGoal:string="";
  dateDebut!:Date;
  dateFin!:Date;
  project!:Project;
  projectID:number=0;
}
